numpy/core/src/multiarray/methods.c

```diff
@@ -2399,6 +2399,12 @@ array_transpose(PyArrayObject *self, PyObject *args)
     return ret;
 }
 
+static PyObject *
+array_matrix_transpose(PyArrayObject *self)
+{
+    return PyArray_MatrixTranspose(self);
```

Comment:
Does this need to be explicitly cast to `PyObject *`

Reply:
No, since `PyArray_MatrixTranspose` returns `PyObject *`, so it matches.  Otherwise you would see a compiler warning anyway.